SYCL/DeviceArchitecture/device_architecture.cpp

```diff
@@ -0,0 +1,44 @@
+// REQUIRES: opencl-aot, cpu
+
+// TODO: remove w/a with "-D__SYCL_TARGET_INTEL_X86_64__ after bug in compiler
+// driver related to if_architecture_is feature is resolved
+// RUN: %clangxx -fsycl -fsycl-targets=intel_gpu_skl,spir64_x86_64 -D__SYCL_TARGET_INTEL_X86_64__ %s -o %t.out
+// RUN: %t.out
+
+#include <sycl/ext/intel/experimental/device_architecture.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::intel::experimental;
+
+int main() {
+  std::vector<int> vec(2);
+  {
+    buffer<int> buf(vec.data(), vec.size());
+
+    queue cpu_queue(cpu_selector_v);
+
+    cpu_queue.submit([&](handler &cgh) {
+      auto acc = buf.get_access<access::mode::read_write>(cgh);
+      cgh.single_task([=]() {
+        if_architecture_is<architecture::x86_64>([&]() {
+          acc[0] = 1;
+        }).otherwise([&]() { acc[0] = 0; });
+      });
+    });
+
+    cpu_queue.submit([&](handler &cgh) {
+      auto acc = buf.get_access<access::mode::read_write>(cgh);
+      cgh.single_task([=]() {
+        if_architecture_is<architecture::intel_gpu_dg1>([&]() {
+          acc[1] = 0;
+        }).otherwise([&]() { acc[1] = 2; });
+      });
+    });
+  }
+
+  assert(vec[0] == 1);
+  assert(vec[1] == 2);
+
+  return 0;
+}
```

Comment:
Some more things that should be tested:

* Test case when more than one `architecture` template parameter is passed to `if_architecture_is`.
* Test the `else_if_architecture_is` function.

Reply:
Thanks, added in https://github.com/intel/llvm-test-suite/pull/1322/commits/8170b069b1f1fd44104784e58b55c80d74113a75